src/gui/SubWindow.cpp

```diff
@@ -26,42 +26,245 @@
 
 #include "SubWindow.h"
 
+#include <QMdiArea>
 #include <QMoveEvent>
 #include <QResizeEvent>
 #include <QWidget>
 
+#include "embed.h"
+#include <QDebug>
+
 
 SubWindow::SubWindow(QWidget *parent, Qt::WindowFlags windowFlags)
   : QMdiSubWindow(parent, windowFlags)
 {
 	// initialize the tracked geometry to whatever Qt thinks the normal geometry currently is.
 	// this should always work, since QMdiSubWindows will not start as maximized
-	m_trackedNormalGeom = normalGeometry();
+  	m_trackedNormalGeom = normalGeometry();
+
+    // inits the colors
+    m_activeColor = Qt::SolidPattern;
+    m_textShadowColor = Qt::black;
+  	m_borderColor = Qt::black;
+
+    //close, minimize, maximize and restore(after minimize) buttons
+    m_closeBtn = new QPushButton( embed::getIconPixmap( "close" ), QString::null, this );
+    m_closeBtn->resize(17, 17);
+    m_closeBtn->setFocusPolicy(Qt::NoFocus);
+    m_closeBtn->setToolTip(tr("Close"));
+    connect(m_closeBtn, SIGNAL(clicked(bool)),this,SLOT(close()) );
+
+    m_maximizeBtn = new QPushButton( embed::getIconPixmap( "maximize" ), QString::null, this );
+    m_maximizeBtn->resize(17, 17);
+    m_maximizeBtn->setFocusPolicy(Qt::NoFocus);
+    m_maximizeBtn->setToolTip(tr("Maximize"));
+    connect(m_maximizeBtn, SIGNAL(clicked(bool)),this,SLOT(showMaximized()));
+
+    m_minimizeBtn = new QPushButton( embed::getIconPixmap( "minimize" ), QString::null, this );
+    m_minimizeBtn->resize(17, 17);
+    m_minimizeBtn->setFocusPolicy(Qt::NoFocus);
+    m_minimizeBtn->setToolTip(tr("Minimize"));
+    connect(m_minimizeBtn, SIGNAL(clicked(bool)),this,SLOT(showMinimized()));
+
+    m_restoreBtn = new QPushButton( embed::getIconPixmap( "restore" ), QString::null, this );
+    m_restoreBtn->resize(17, 17);
+    m_restoreBtn->setFocusPolicy(Qt::NoFocus);
+    m_restoreBtn->setToolTip(tr("Restore"));
+    connect(m_restoreBtn, SIGNAL(clicked(bool)),this,SLOT(showNormal()));
+
+    // QLabel for window title and shadow effect
+    m_shadow = new QGraphicsDropShadowEffect();
+    m_shadow->setColor(m_textShadowColor);
+    m_shadow->setBlurRadius(3);
+    m_shadow->setXOffset(1);
+    m_shadow->setYOffset(1);
+
+    m_windowTitle = new QLabel(this);
+    m_windowTitle->setFocusPolicy(Qt::NoFocus);
+    m_windowTitle->setGraphicsEffect(m_shadow);
+}
+
+
+
+
+void SubWindow::paintEvent(QPaintEvent *)
+{
+    QPainter p(this);
+    const int titleBarHeight = 24;
+    QRect rect(0, 0, width(), titleBarHeight);
+    bool isActive = SubWindow::mdiArea()->activeSubWindow() == this;
+
+    p.fillRect(rect, isActive ? activeColor() : p.pen().brush() );
+
+    // window border
+    p.setPen(borderColor());
+
+    // bottom, left, and right lines
+    p.drawLine( 0, height() - 1, width(), height() - 1 );
+    p.drawLine( 0, titleBarHeight, 0, height() - 1 );
+    p.drawLine( width() - 1, titleBarHeight, width() - 1, height() - 1 );
+
+    //window icon
+    QPixmap winicon(widget()->windowIcon().pixmap(QSize(17,17)));
```

Comment:
same applies here as in https://github.com/LMMS/lmms/pull/2516/commits/11c25a62187cc2291d015feca6818e1f9957b5f8#r60984624  , a `const int` to replace the instances of `17` should be made.